### ISSUE TITLE
Dose the latest version only support 1.4 firmware？

### ISSUE THREAD

**Issue by hualazimo7** [lesserkuma/FlashGBX#107]
My DIY dumper's firmware is rw1.3 L1, flashgbx suggests "please update fw 1.4".

**Comment by lesserkuma** [comment]
I don't know about DIY dumpers, but the original GBxCart RW v1.3 is supported as of FlashGBX v3.37. Some features may still require the v1.4 hardware to work, though.

**Comment by mattlawer** [comment]
Bootleg GBFlash are CH579M-based and not compatible with the embedded fw_GBFlash firmware for CH340. Might be useful to embed the CH579M-based firmware and modify the FirmwareUpdater to allow updating those GBFlash.

**Comment by Lesserkuma** [comment]
Original GBFlash uses a CH579M, not CH340.
I do not own a bootleg GBFlash, but as far as I have heard, they have stripped out the firmware update feature intentionally.

**Comment by mattlawer** [comment]
Yes I got a bootleg one from TicKnot Store, pressing the U22 button at boot doesn't make it boot in DFU mode like the original one. The bootleg one might be based on first versions of GBFlash public schematics that were using an ATmega, the new (original one) seem to be based on STM micro. [Schematics for older, ATmega based versions were available publicly](https://www.reddit.com/r/Gameboy/comments/1i2zo8z/comment/m7sefsn/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)

**Comment by Lesserkuma** [comment]
Hmm, the U22 button is not even used on a GBFlash v1.3. It’s only used for v1.2 and older that are still running the “L0” firmware.

**Comment by visallard** [comment]
Is it possible to bypass the regular update process with FlashGBX using a chip-specific flashing method ? My GBFlash (that looks fairly legit, but you never know) won't update, but I can force it to boot in ISP mode and I would be likely able to directly flash the fw.bin provided with WCHISPStudio ; maybe there is a license bit hidden somewhere that would brick it ? Seems that the firmware on it is read-protected so I can't back it up before committing.

**Comment by Lesserkuma** [comment]
Does your GBFlash have a component soldered at “X1” on the circuit board? If yes, it’s probably a fake.
Unfortunately, I have no idea how to otherwise flash the fw.bin, and I do not own a fake GBFlash myself.

**Comment by visallard** [comment]
Ah shucks, it does ; it's a 32KHz oscillator, I guess they just skip the factory calibration of the internal oscillator. That's a pretty damn copy in any case ; apart from that it looks like a 1:1 replica.

Guess I'll leave it be for now, maybe I'll bite the bullet some day ! I've read the CH579 datasheet and it doesn't look like there is much difference between the IAP and ISP update process.

Edit : BTW, it is totally out of scope for this project, but it would be greatly beneficial to have a chip reflash emergency procedure, as I've had multiple microcontrollers losing/corrupting their program memory before. Saving an otherwise good board is always a better alternative to replacing <3

**Comment by Lesserkuma** [comment]
Yeah I agree 👍 
But as I’m not the creator of the GBFlash hardware device, I don’t know any more about it either 😅
